library/pdf_creator.class.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * PdfCreator class
+ *
+ * @package OpenEMR
+ * @link    http://www.open-emr.org
+ * @author  Jerry Padgett <sjpadgett@gmail.com>
+ * @copyright Copyright (c) 2017 Jerry Padgett <sjpadgett@gmail.com>
+ * @license https://www.gnu.org/licenses/agpl-3.0.en.html GNU Affero General Public License 3
+ */
+use Knp\Snappy\Pdf;
+use Symfony\Component\Debug\ExceptionHandler;
+
+class PdfCreator
+{
+
+    private $binaryPath;
+
+    private $tempPath;
+
+    private function getBinaryPath()
+    {
+        global $webserver_root;
+        // This will not necessarily reflect actual machine bus width but php bus size.
+        $intsize = strlen(decbin(~ 0));
+        if (empty(strstr(php_uname("m"), '64'))) {
+            $bit = "32bit";
+        } else {
+            $bit = "64bit";
+        }
+        try {
```

Comment:
I looked at https://github.com/KnpLabs/snappy and I see what you mean. Looks like the main source is here: https://wkhtmltopdf.org/
Which is then built into binaries for a bunch of different os's in different places.
Quick question here; will the binary for example in ubuntu be different than other linux flavors?

This strikes as our very good first case to create a simple new github repo (wkhtmltopdf) which can then be loaded/updated into OpenEMR via composer(thus all work is done in this wkhtmltopdf upstream repo and then just brought into the OpenEMR codebase via composer). So will basically store all the possible binaries for this here (also then gives way for users to even build binaries for other os's and then contribute them to this custom wkhtmltopdf repo).